### PR TITLE
internally document functions with JSDoc

### PR DESCRIPTION
**Pull request by staltz**
Added JSDoc which shows to many IDEs out there, and clarified the meaning of several parameters.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -39,29 +44,50 @@ function decode(bmsg) {
...
-    bfe.encodeBendyButt(msg.signature),
+    bfe.encodeBendyButt(msgVal.signature),
   ])
 }
 
-// returns a classic compatible json object
+/**
+ * Creates a new bendy-butt message that can be appended to a bendy-butt feed.
+ *
+ * FIXME: this method does not seem to belong to `ssb-bendy-butt` because it is
```
Comment (mycognosist) [inline index.js:77]: I was also wondering about this. I have been holding-off on implementing this in Rust for the moment. I would tend towards having `ssb-meta-feeds` import the necessary encoders (BFE and Bendy) and performing the object creation itself.

File: index.js [hunk 2]
```diff
@@ -39,29 +44,50 @@ function decode(bmsg) {
...
+ * Encode a database "msg value" to a bendy-butt msg in a Buffer.
+ *
+ * @param {Object} msgVal an object compatible with ssb/classic `msg.value`
+ * @returns {Buffer} a bendy-butt message encoded with `bencode`
+ */
+function encode(msgVal) {
   const contentSection =
-    typeof msg.content === 'string' && msg.content.endsWith('.box2')
-      ? bfe.encodeBendyButt(msg.content)
+    typeof msgVal.content === 'string' && msgVal.content.endsWith('.box2')
```
Comment (mycognosist) [inline index.js:55]: `msg` -> `msgVal` smart move.

**Review by mycognosist — APPROVED** [review]
JSDoc clarifications, especially concerning parameters and return types, are a great addition.

File: index.js (hunk 1 above)
Comment (staltz) [reply index.js:77]: I'm working on something to address this. I'm repurposing `create` to `encodeNew` and I'm abstracting away some details, and I intend to update the [bendy-butt-spec](https://github.com/ssb-ngi-pointer/bendy-butt-spec) to not mention anything metafeeds or subfeeds related.

**Comment by staltz** [comment]
Thanks for taking a look!

File: index.js (hunk 1 above)
Comment (mycognosist) [reply index.js:77]: > Technically people could use bendy-butt for purposes other than metafeeds, right?

Absolutely. Decouple the tools so they can be useful in many contexts.

**Pull request merged**